control-plane/subcommand/inject-connect/command.go

```diff
@@ -464,14 +470,34 @@ func (c *Command) Run(args []string) int {
 			ConsulAPITimeout:              c.http.ConsulAPITimeout(),
 		}})
 
+	if c.flagEnableWebhookCAUpdate {
+		err := c.updateWebhookCABundle(ctx)
```

Comment:
The Run function does not have many tests around except flag validation.  I tried to add tests, but the fact that we set LeaderElection to true causes the leader election code to get the namespace from the `/var/run/secrets/kubernetes.io/serviceaccount/namespace` in [this section](https://github.com/kubernetes-sigs/controller-runtime/blob/f46919744bee01060c9084a285e049afffd38c9d/pkg/leaderelection/leader_election.go#L115) that fails in tests.  I have tried creating this file during the unit test but creating anything in `/var/run` fails with secrets errors.

Some options:
- create an `enable-leader-election` flag that defaults to true and we can set it to false in tests
- add `leaderElection` to the command similar to `clientSet` and do similarly as above where we set it to false in tests
- make `updateWebhookCABundle()` public and test that.  Don't worry about the logic that just looks at the flag and calls this when it is true.
- do nothing.  (Related to the last option above making `updateWebhookCABundle()` public.  It just reads a file and calls `mutatingwebhookconfiguration.UpdateWithCABundle()` which already is tested pretty well. )

Thoughts?

Reply:
I actually tried option #2 above and made changes that look a lot like `changing my production code to add in test code` which I have mixed feelings about.  Once it got to the point that the Controller manager was started, the execution of the test just hung as this separate process ran.  It never came back and the test timed out.  I feel like this may not be worth the effort to make sure that mutatingwebhookconfiguration.UpdateWithCABundle() gets called since we have unit tests that assert that works properly and we have acceptance tests that this works as a whole.